inference/core/workflows/core_steps/analytics/time_in_zone/v1.py

```diff
@@ -0,0 +1,143 @@
+from typing import Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import supervision as sv
+from pydantic import ConfigDict, Field
+from typing_extensions import Literal, Type
+
+from inference.core.workflows.execution_engine.entities.base import (
+    OutputDefinition,
+    VideoMetadata,
+    WorkflowImageData,
+)
+from inference.core.workflows.execution_engine.entities.types import (
+    INSTANCE_SEGMENTATION_PREDICTION_KIND,
+    LIST_OF_VALUES_KIND,
+    OBJECT_DETECTION_PREDICTION_KIND,
+    STRING_KIND,
+    StepOutputImageSelector,
+    StepOutputSelector,
+    WorkflowImageSelector,
+    WorkflowParameterSelector,
+    WorkflowVideoMetadataSelector,
+)
+from inference.core.workflows.prototypes.block import (
+    BlockResult,
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+
+OUTPUT_KEY: str = "time_in_zone"
+TYPE: str = "PerspectiveCorrection"
+SHORT_DESCRIPTION = "Track duration of time spent by objects in zone"
+LONG_DESCRIPTION = """
+The `TimeInZoneBlock` is an analytics block designed to measure time spent by objects in a zone.
+The block requires detections to be tracked (i.e. each object must have unique tracker_id assigned,
+which persists between frames)
+"""
+
+
+class TimeInZoneManifest(WorkflowBlockManifest):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "name": "Time in zone",
+            "version": "v1",
+            "short_description": SHORT_DESCRIPTION,
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "analytics",
+        }
+    )
+    type: Literal["roboflow_core/time_in_zone@v1", "TimeInZone"]
+    image: Union[WorkflowImageSelector, StepOutputImageSelector] = Field(
+        title="Image",
+        description="The input image for this step.",
+        examples=["$inputs.image", "$steps.cropping.crops"],
+    )
+    metadata: WorkflowVideoMetadataSelector
+    detections: StepOutputSelector(
+        kind=[
```

Comment:
we require tracking for that block and put Union[`OBJECT_DETECTION_PREDICTION_KIND`, `INSTANCE_SEGMENTATION_PREDICTION_KIND`] kind here, whereas tracking only applies for `OBJECT_DETECTION_PREDICTION_KIND`